_posts/2024-01-19-yannick-horrix-opentofu.md

```diff
@@ -0,0 +1,160 @@
+Terraform currently holds the advantage when it comes to community support. 
+The Terraform community is vast, offering a large breadth of resources. 
+This means that users have more readily accessible examples at their disposal and can find solutions to their problems more quickly.
+
+OpenTofu, on the other hand, is a growing tool with an expanding community. 
+Although not as large as Terraform's, its community is rapidly expanding as more users move to OpenTofu. 
+This will provide ever increasing support and resources for the tool in the future.
+## Maturity and Stability
+Terraform is proven as a mature, stable and extremely popular orchestration tool capable of handling enterprise-grade infrastructure deployments.
+OpenTofu, as a fork of Terraform, is a newer tool with a relatively newer and may be more susceptible to bugs and instability. However, the “OpenTofu” community is actively addressing these issues, and its stability is expected to improve with time.
+## Summary
+To summarize, the following table captures the comparison between the two tools.
+
+| Feature | OpenTofu | Terraform |
+|----------|----------|----------|
+| Features | Similar to Terraform, with some improvements | Large featureset |
+| Configuration | Declarative, easier to use | Procedural, more complex |
+| Licensing | Mozilla Public License (MPL), open-source | Business Software License (BUSL), open-source with restrictions |
+| Pricing | Free to use | Free to use, with some restrictions for commercial use |
+| Community | Smaller, but expanding | Large |
+| Maturity | New | Mature |
+
+# How to Migrate?
+OpenTofu promises a seamless migration for Terraform users. 
+As both OpenTofu 1.6 and Terraform 1.6 are compatible, the migration process is relatively straightforward. 
+However, it is advised to have a disaster recovery plan in place as the migration still poses a non-trivial change to the architecture. 
+The migration process is detailed in the [OpenTofu Migration Guide](https://opentofu.org/docs/intro/migration/){:target="_blank" rel="noopener noreferrer"}.
+
+In summary, the process looks like this:
+1. Apply all changes with Terraform
+
+   `terraform apply`
+2. [Install OpenTofu](https://opentofu.org/docs/install/){:target="_blank" rel="noopener noreferrer"}
+   
+    Test if you can successfully execute the `tofu` command:
+
+    `tofu --version`
+3. Back up your state file
+   
+   If you are using a local state file, you can simply make a copy of your `terraform.tfstate` file in your project directory.
+
+   If you are using a remote backend such as an S3 bucket, make sure that you follow the backup procedures for the backend and that you exercise the restore procedure at least once. 
+4. Initialize OpenTofu
+
+   `tofu init`
+5. Inspect the Plan
+
+   `tofu plan`
+6. Test out a small change
+
+   `tofu apply`
+
+And that's it.
+Migration complete!
+
+In order to rollback, it suffices to create a backup of the new state file and perform
```

Comment:
This rollback scenario only works if you revert between the same versions (1.6.0)